images/Dockerfile

```diff
@@ -39,28 +37,35 @@ ENV RUNNER_MANUALLY_TRAP_SIG=1
 ENV ACTIONS_RUNNER_PRINT_LOG_TO_STDOUT=1
 ENV ImageOS=ubuntu22
 
-# 'gpg-agent' and 'software-properties-common' are needed for the 'add-apt-repository' command that follows
-RUN apt update -y \
-    && apt install -y --no-install-recommends sudo lsb-release gpg-agent software-properties-common curl jq unzip \
-    && rm -rf /var/lib/apt/lists/*
-
-# Configure git-core/ppa based on guidance here:  https://git-scm.com/download/linux
-RUN add-apt-repository ppa:git-core/ppa \
-    && apt update -y \
-    && apt install -y --no-install-recommends git
+# Copy PPAs from build stage
+COPY --from=build /etc/apt/sources.list.d/*.list /etc/apt/sources.list.d/
+COPY --from=build /etc/apt/trusted.gpg.d/*.gpg /etc/apt/trusted.gpg.d/
 
-RUN adduser --disabled-password --gecos "" --uid 1001 runner \
-    && groupadd docker --gid 123 \
-    && usermod -aG sudo runner \
-    && usermod -aG docker runner \
-    && echo "%sudo   ALL=(ALL:ALL) NOPASSWD:ALL" > /etc/sudoers \
-    && echo "Defaults env_keep += \"DEBIAN_FRONTEND\"" >> /etc/sudoers
+# Install docker and cli-plugins as a copy from docker:cli image
+COPY --from=docker-cli --chown=root:root --chmod=0555 /usr/local/bin/docker /usr/bin/docker
+COPY --from=docker-cli --chown=root:root --chmod=0555 /usr/local/libexec/docker/cli-plugins/docker-buildx /usr/local/libexec/docker/cli-plugins/docker-buildx
 
-WORKDIR /home/runner
+# Install a sane set of base utilities
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends sudo lsb-release gpg-agent software-properties-common curl git jq unzip \
+ && apt-get clean all \
+ && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
```

Comment:
why remove the lists here? wouldn't that mean that the next `apt-get update` has to download all of them again?

Reply:
I was optimizing for size. Yes, it has to download again, but the cache hit rate on lists is very low anyway. I'll rework this section.

Reply:
Upstream image (mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy) also ships with an empty /var/lib/apt/lists directory. I like it this way. Willing to discuss though.

Reply:
Maybe there is a misunderstanding. I'm not against `apt-get clean all && rm -rf /var/lib/apt/lists/* /var/cache/apt/*` – I was just wondering, why you did it multiple times in a row and not just once at the end.

Reply:
No reason. That's an easy fix-up. It only has an effect at build time. I'll move the upgrade into the top block and save a layer at the same time. But first, lunch.

Reply:
All done. Have a look.